core/lib/basic_types/src/errors.rs

```diff
@@ -0,0 +1,62 @@
+use std::fmt;
+
+/// Represents an error in a stoppable task: either an internal / fatal error, or a task getting stopped
+/// after receiving a stop signal.
+///
+/// The [`Error`](std::error::Error) trait is intentionally not implemented for this enum to force users
+/// to treat the `Stopped` variant with care. Depending on the application, it may not be an error.
+/// Use the [`try_stoppable!` macro](crate::try_stoppable) to handle this variant by early-returning `Ok(())`.
+#[derive(Debug)]
+pub enum OrStopped<E = anyhow::Error> {
```

Comment:
Nite: I like that we move away from calling this "interruption", I would also not use "signal" in the docs as that is easy to confuse with POSIX signals

Reply:
Re POSIX signals, I've thought about the same thing 🙃 As a counterpoint, rn in node binaries, stopping *is* initiated as a result of receiving a SIGINT / SIGTERM. What would be the preferred terms then? Something like "stop request"?

Reply:
> stopping is initiated as a result of receiving a SIGINT / SIGTERM

Also if one of the tasks exits prematurely (i.e. panics), no?

> What would be the preferred terms then? Something like "stop request"?

Yeah I think "stop request" is okay since we are going with the "stopped" language. For reference tokio's [`CancellationToken`](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html) calls it "signal a cancellation request" so we could maybe rephrase to "or a task getting stopped after being signaled to do so". In any case no strong opinion here.